src/python/pants/reporting/reporting.py

```diff
@@ -112,6 +112,14 @@ def initialize(self, run_tracker, all_options, start_time=None):
       raise ValueError(
         "Flags zipkin-trace-id and zipkin-parent-id must both either be set or not set."
       )
+    if trace_id and len(trace_id) != 16 and len(trace_id) != 32:
+      raise ValueError(
+        "Value of the flag zipkin-trace-id must be a 16-character or 32-character hex string."
```

Comment:
Would it be useful to check whether the string is a hex string (I'm assuming it should be `[0-9a-f]`)? Would it be useful to turn this into a regex or helper method? Checking with a regexp could be `re.match(r'[0-9a-f]{16}|[0-9a-f]{32}', trace_id)`.

Reply:
It would also be good to include the passed value so that it's easier to track down in debugging.